File: apps/server/src/utils/api-response.ts
export interface ApiResponse<T> {
    code: number;
    msg: string;
    data: T;
}

export function createApiResponse<T>(code: number, msg: string, data: T): ApiResponse<T> {
    return { code, msg, data };
}

export function createSuccessResponse<T>(data: T, msg = '成功'): ApiResponse<T> {
    return createApiResponse(200, msg, data);
}

File: apps/server/app/schemas/api_response.py
from typing import Generic, TypeVar

from pydantic import BaseModel


DataT = TypeVar("DataT")


class ApiResponse(BaseModel, Generic[DataT]):
    code: int
    msg: str
    data: DataT


class ErrorData(BaseModel):
    requestId: str
    errorCode: str | None = None


def success_response(data: DataT, message: str = "成功") -> ApiResponse[DataT]:
    return ApiResponse(code=200, msg=message, data=data)
